src/registry/routes/component.js

```diff
@@ -20,12 +20,8 @@ module.exports = function(conf, repository){
         res.errorDetails = result.response.error;
       }
 
-      if (!_.isEmpty(result.response.headers)) {
-        res.set(result.response.headers);
-        
-        if (req.method === 'GET') {
-          delete result.response.headers;
-        }
```

Comment:
This is not actually needed as `component.js` is used only by the GET endpoint. Also `delete` can be not great for performances so I revert the strategy (to not set it in the response and then to make it part of the POST individual result)